flocker/control/httpapi.py

```diff
@@ -796,6 +799,38 @@ def delete_container_configuration(self, name):
         # Didn't find the application:
         raise CONTAINER_NOT_FOUND
 
+    @app.route("/configuration/_compose", methods=['POST'])
+    # Undocumented for now since this is temporary, private endpoint for
+    # flocker-deploy.
+    @structured(
+        inputSchema={
+            '$ref':
+            '/v1/endpoints.json#/definitions/configuration_compose'
+        },
+        outputSchema={},
+        schema_store=SCHEMAS
+    )
+    def replace_configuration(self, applications, deployment):
+        """
+        Replace the existing configuration with one given by flocker-deploy
+        command line tool.
+
+        :param applications: Configuration in Flocker-native or
+            Fig/Compose format.
+
+        :param deployment: Configuration of which applications run on
+            which nodes.
+        """
+        try:
+            configuration = FigConfiguration(applications)
+            if not configuration.is_valid_format():
+                configuration = FlockerConfiguration(applications)
+            return self.persistence_service.save(model_from_configuration(
```

Comment:
You noticed that this changes the behavior of `flocker-deploy` so it completes before the change has been deployed.  Perhaps that change could be avoided with a poll loop here that only provides a result when state and configuration are sufficiently similar?  Maybe we don't have an API that makes comparing a `DeploymentState` to a `Deployment` very easy, yet, though...

On one hand this doesn't really matter a lot because we're definitely going to drop `flocker-deploy` completely before 1.0.  On the other hand, what if I'm wrong and we fail to get rid of it in time? :/